Several dependencies have needlessly restrictive version requirements, which
causes build failures against more up-to-date Void packages.

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -11,13 +11,13 @@
 build-backend = 'mesonpy'
 requires = [
     # Reason for `<`: future-proofing (0.14.0 released and known to work)
-    "meson-python>=0.12.1,<0.15.0",
+    "meson-python>=0.12.1",
     # Reason for `<`: we want to build the 1.11.x releases with 0.29.x (too many changes in 3.0)
-    "Cython>=0.29.35,<3.0",   # when updating version, also update check in meson.build
+    "Cython>=0.29.35",
     # Reason for `<`: future-proofing (2.11.1 is the most recent version)
-    "pybind11>=2.10.4,<2.11.1",
+    "pybind11>=2.10.4",
     # Reason for `<`: future-proofing (0.14.0 released and known to work)
-    "pythran>=0.12.0,<0.15.0",
+    "pythran>=0.12.0",
 
     # NumPy dependencies - to update these, sync from
     # https://github.com/scipy/oldest-supported-numpy/, and then
